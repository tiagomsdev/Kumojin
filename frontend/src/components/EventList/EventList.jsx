import React from "react";
import { useEffect } from "react";
import Table from "@mui/material/Table";
import TableBody from "@mui/material/TableBody";
import TableCell from "@mui/material/TableCell";
import TableContainer from "@mui/material/TableContainer";
import TableHead from "@mui/material/TableHead";
import TableRow from "@mui/material/TableRow";
import Paper from "@mui/material/Paper";
import axios from "axios";
import { EventManagerUrl } from "../../common/constants";
import { parseCookies } from "nookies";

function EventList({ eventList, handleEventList }) {
    
  useEffect(() => {
    const cookies = parseCookies();
    const headers = {
      "Content-Type": "application/json",
      Authorization: `Bearer ${cookies["USER_TOKEN"]}`,
    };
    axios
      .get(EventManagerUrl, { headers })
      .then(function (response) {
        handleEventList(response.data);
        console.log(response);
      })
      .catch(function (error) {
        console.log(error);
      });
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <TableContainer component={Paper}>
      <Table sx={{ minWidth: 800 }} aria-label="simple table">
        <TableHead data-testid="eventList-tableHead">
          <TableRow>
            <TableCell>Nom de l'événement</TableCell>
            <TableCell align="left">Description de l'événement</TableCell>
            <TableCell align="left">Date/Heure de début</TableCell>
            <TableCell align="left">Date/Heure de fin</TableCell>
          </TableRow>
        </TableHead>
        <TableBody data-testid="eventList-tablebody">
          {eventList.map((row, idx) => (
            <TableRow
              key={idx}
              sx={{ "&:last-child td, &:last-child th": { border: 0 } }}
            >
              <TableCell data-testid="eventList-tableCell" component="th" scope="row">
                {row.name}
              </TableCell>
              <TableCell align="left">{row.description}</TableCell>
              <TableCell align="left">{row.startDate}</TableCell>
              <TableCell align="left">{row.endDate}</TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}

export default EventList;
