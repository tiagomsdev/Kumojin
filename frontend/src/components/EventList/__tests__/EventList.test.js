import {render, screen, cleanup} from '@testing-library/react';
import EventList from "../../EventList/EventList";
import renderer from "react-test-renderer";

test('should render EventList component', () => {
    
    const data = [{idEvent: '600dc62d-f31a-45db-8b87-537a0970efcd', name: 'test1', description: 'test1', startDate: 'Nov 11th 2021, 6:30 pm', endDate: 'Nov 11th 2021, 6:30 pm'},
                 {idEvent: '0a8fa35e-d0ad-4a0d-8275-690805c62409', name: 'test2', description: 'test2', startDate: 'Nov 11th 2021, 6:30 pm', endDate: 'Nov 11th 2021, 8:19 pm'}];
    
    render(<EventList eventList={data}/>);

    const tableElement = screen.getByTestId('eventList-tableHead');
    const tableBodyElement = screen.getByTestId('eventList-tablebody');


    expect(tableElement).toBeInTheDocument();
    expect(tableBodyElement).toBeInTheDocument();
    
});

test('should matches EventList snapshot', () =>{
    const data = [{idEvent: '600dc62d-f31a-45db-8b87-537a0970efcd', name: 'test1', description: 'test1', startDate: 'Nov 11th 2021, 6:30 pm', endDate: 'Nov 11th 2021, 6:30 pm'},
                 {idEvent: '0a8fa35e-d0ad-4a0d-8275-690805c62409', name: 'test2', description: 'test2', startDate: 'Nov 11th 2021, 6:30 pm', endDate: 'Nov 11th 2021, 8:19 pm'}];

    const eventListTree = renderer.create(<EventList eventList={data}/>).toJSON();
    expect(eventListTree).toMatchSnapshot();
});