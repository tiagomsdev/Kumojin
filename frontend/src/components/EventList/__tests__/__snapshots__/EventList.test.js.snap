// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should matches EventList snapshot 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiTableContainer-root css-11xur9t-MuiPaper-root-MuiTableContainer-root"
>
  <table
    aria-label="simple table"
    className="MuiTable-root css-nv4u2v-MuiTable-root"
    role={null}
  >
    <thead
      className="MuiTableHead-root css-15wwp11-MuiTableHead-root"
      data-testid="eventList-tableHead"
      role={null}
    >
      <tr
        className="MuiTableRow-root MuiTableRow-head css-1q1u3t4-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Nom de l'événement
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Description de l'événement
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Date/Heure de début
        </th>
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-head MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ygcj2i-MuiTableCell-root"
          scope="col"
        >
          Date/Heure de fin
        </th>
      </tr>
    </thead>
    <tbody
      className="MuiTableBody-root css-apqrd9-MuiTableBody-root"
      data-testid="eventList-tablebody"
      role={null}
    >
      <tr
        className="MuiTableRow-root css-34nofg-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          data-testid="eventList-tableCell"
          scope="row"
        >
          test1
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          test1
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Nov 11th 2021, 6:30 pm
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Nov 11th 2021, 6:30 pm
        </td>
      </tr>
      <tr
        className="MuiTableRow-root css-34nofg-MuiTableRow-root"
        role={null}
      >
        <th
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
          data-testid="eventList-tableCell"
          scope="row"
        >
          test2
        </th>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          test2
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Nov 11th 2021, 6:30 pm
        </td>
        <td
          aria-sort={null}
          className="MuiTableCell-root MuiTableCell-body MuiTableCell-alignLeft MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root"
        >
          Nov 11th 2021, 8:19 pm
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
